### ISSUE TITLE
Pygame 2.0 compatability

### ISSUE THREAD

**Issue by bitcraft** [Tuxemon/Tuxemon#547]
Tuxemon is currently not compatible with pygame 2.  This issue tracks progress to that goal.  Please add comments related to pygame 2 compatability here.  If you find an issue please create a new issue.

**Comment by bitcraft** [comment]
Currently keyboard and input handling breaks for pygame 2 and the game will not start in the world state.  Coincidently, my input refactor does work with pygame2 and will unblock more testing.

**Comment by kerizane** [comment]
How are you testing this? I thought pygame 2 wasn't out yet.

**Comment by bitcraft** [comment]
```pip install pygame==2.0.0.dev6```

Its not 100% ready now, but its mostly working, with a lot of bugs.  Tuxemon is playable using it, but there are some texture/surface transparency issues.  Not sure if it is because of our code or bugs with pygame.

**Comment by kerizane** [comment]
Players eyes are transparent.

**Comment by bitcraft** [comment]
hahaha, yeah, the dead, transparent eyes really look bad.

**Comment by bitcraft** [comment]
I just added a new issue and linked it with the pygame 2 milestone.  I don't recall anything else that was obviously broken though.  Im surprised/happy that none of our dependencies need updating to "just work"...though there are obviously graphical glitches.

**Comment by Qiangong2** [comment]
Maybe it's due to some other issue, but Tuxemon does not boot at all on the GameShell with pygame 2.0. However, with standard pygame it runs fine.

Does pygame 2 use more memory by chance? My GameShell board only has 512mb of RAM

**Comment by bitcraft** [comment]
@Qiangong2 I don't think it uses more memory.  512mb should be fine.  Pygame 2 is implemented using most of the same code as pygame 1.x.  Do you get a traceback?  Can you use strace to get more info?

**Comment by kerizane** [comment]
Random game stutters with pygame==2.0.0.dev6, does not affect pygame==2.0.0.dev4

When I walk around the game will randomly freeze for roughly a second. Happens on all maps.

**Comment by Qiangong2** [comment]
@bitcraft the new 0.5 OS works fine now with Tuxemon. It must have just been a quirk with 0.4. However, I do still see the random freezes with pygame 2.0

**Comment by bitcraft** [comment]
Pygame 2 is officially supported and tested
